### DISCUSSION TITLE
SSRS Integration

### DISCUSSION THREAD

**Discussion by a-ashok** [abpframework/abp#14342] [Q&A]
Are there any ways to integrate SSRS (Sql Server Reporting Services) into ABP for reporting needs.

**Comment by tinonetic** [comment]
Howt exactly do you want to integrate? 

Simplest is have a link that leads to the SSRS Report Viewer, then the user can navigate there
Or pass parameters to the SSRS report URL

**Reply by a-ashok** [reply]
Hi i want to integrate the report viewer inside. pass paramaters @ runtime and load the report within the cshtml page, just as we do for ASP.NET MVC. is this workable?

**Comment by sonrai-LLC** [comment]
The 2 primary ways (and 1 legacy way) to integrate SSRS:

**SSRS URL Access "API"** (not really an API, but you can pass report parameters and do some view customization, this basically returns the report viewer control (with the Print button, etc, and you can use querystring values to control the user's view*))): https://learn.microsoft.com/en-us/sql/reporting-services/url-access-ssrs?view=sql-server-ver16

**SQL Server 2017 Reporting Services REST API** (this offers fine grained control, so you can for instance, GET a collection of report schedules, or POST to create a session to authenticate users, almost anything that can be done in /reports portal is using the API- https://app.swaggerhub.com/apis/microsoft-rs/SSRS/2.0

**SSRS SOAP API** (Legacy API, SOAP and XML-based) https://learn.microsoft.com/en-us/sql/reporting-services/report-server-web-service/accessing-the-soap-api?view=sql-server-ver16 This may be more familiar for some developers and actually provides a lot more functionality than the REST API, indeed the latter calls into the legacy former at the ReportingServicesService level.


Not sure of your req's but I have a project (well, projects, only the latter is being actively maintained) that you can customize that may help you. https://github.com/sonrai-LLC/ExtRSAuth and https://github.com/sonrai-LLC/ExtRS



*one problem for many orgs is, you first have to authenticate to a Windows domain; not exactly a selling point for SQL Server on Linux (you can use **ExtRSAuth** to implement custom//non-Windows auth on the report server, until MS has a real solution), but... I digress from your question. If any other questions SSRS or SSRS Auth-related, lmk; I may be able to help.
